### ISSUE TITLE
[Bug] Granular Read Permissions do not translate into complete Read permission

### ISSUE THREAD

**Issue by RicardoMonteiroSimoes** [thingsboard/thingsboard#9370]
**Describe the bug**

Currently I am setting up a Customer and users for him. One thing we are doing is dedicated dashboards for each group of users they have.

I have created a role to read only dashboards, and separate one to limit view of certain dashboards.
The issue here is that unless I give a complete _read_ operation to the role, the user _can not_ load any telemetry on the dashboard:

![image](https://github.com/thingsboard/thingsboard/assets/30594398/fd9fe65f-20d1-4afd-b54a-2d899a0a67c5)

I then tried to add all the granular read operations by themselves, to figure out which one enables the dashboard to load data, but even adding them all doesn't make it work; I have to specifically add the full _read_ operation for it to work.

**Your Server Environment**
    cloud deployment
    ThingsBoard Version
    OS: Linux (ubuntu 20.04)

**To Reproduce**
1. Create User and without any user roles
2. Add a new Role and give it all granular Read operations, as well as Read for widget type
3. Create dashboard and add a device (as an admin)
4. Dashboard will not load data:

![image](https://github.com/thingsboard/thingsboard/assets/30594398/761ef661-7fa3-4635-a014-1dee213d5a5c)

**Expected behavior**
Selecting all granular read operations should be the same as selecting `read`

**Comment by devaskim** [comment]
Did you try `Read` instead of `All` ?

**Comment by RicardoMonteiroSimoes** [comment]
> Not the answer, but did you try `Read` instead of `All` ?

Yes, that doesn't work as well. I have to add all Read variants for it to work.

**Comment by SuminDenis** [comment]
Hi @RicardoMonteiroSimoes,

First of all, I wanna mention that the _Read_ operation doesn't mean that you can read all the information about the device. With _Read_ operation, you can only read the following fields: device name, label, profile, description, etc.

There are several reasons why you cannot see telemetry from the device:

1. You try to read information from the Tenant Device on the Customer level (if the Tenant doesn't share permissions for this device with a particular Customer, you cannot read the information contained in the device.
7. You didn't include Widget Bundle Resource.

We will close this issue due to the long inactivity. If you still need help, please open a new issue with the following information:

1. ThingsBoard version.
2. Information about _Role_ (_Type, Resources, and Operations)_.
8. Customer hierarchy.
9. Device and Dashboard information (Permissions, Owner, Groups)
